src/avfoundation.cs

```diff
@@ -10764,6 +11172,56 @@ interface AVSpeechSynthesizerDelegate {
 		void WillSpeakRangeOfSpeechString (AVSpeechSynthesizer synthesizer, NSRange characterRange, AVSpeechUtterance utterance);
 	}
 
+	enum AVAssetDownloadedAssetEvictionPriority {
+		[NoWatch, NoTV, NoMac, iOS (11, 0)]
+		[Field ("AVAssetDownloadedAssetEvictionPriorityDefault")]
+		Default = 0,
+
+		[NoWatch, NoTV, NoMac, iOS (11, 0)]
+		[Field ("AVAssetDownloadedAssetEvictionPriorityImportant")]
+		Important = 1,
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface AVAssetDownloadStorageManager
+	{
+		[Static]
+		[Export ("sharedDownloadStorageManager")]
+		AVAssetDownloadStorageManager SharedDownloadStorageManager { get; }
+
+		[Export ("setStorageManagementPolicy:forURL:")]
+		void SetStorageManagementPolicy (AVAssetDownloadStorageManagementPolicy storageManagementPolicy, NSUrl downloadStorageUrl);
+
+		[Export ("storageManagementPolicyForURL:")]
+		[return: NullAllowed]
+		AVAssetDownloadStorageManagementPolicy StorageManagementPolicyForUrl (NSUrl downloadStorageUrl);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface AVAssetDownloadStorageManagementPolicy : NSCopying, NSMutableCopying
+	{
+		[Export ("priority")]
+		AVAssetDownloadedAssetEvictionPriority Priority { get; }
+
+		[Export ("expirationDate", ArgumentSemantic.Copy)]
+		NSDate ExpirationDate { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (11,0)]
+	[BaseType (typeof(AVAssetDownloadStorageManagementPolicy))]
+	interface AVMutableAssetDownloadStorageManagementPolicy
+	{
+		[Export ("priority")]
+		AVAssetDownloadedAssetEvictionPriority Priority { get; set; }
+
+		[Export ("expirationDate", ArgumentSemantic.Copy)]
+		NSDate ExpirationDate { get; set; }
```

Comment:
that should have issue a warning ?!?
C# can't override a getter-only property with a get/set
The trick is to define the non-mutable API like this:
`NSDate ExpirationDate { get; [NotImplemented] set; }`
so it can be overridden correctly

Reply:
That's not correct. This `[NotImplemented]` is needed on the *non mutable* type. The removal here is fine but the original issue remains.